setup.py

```diff
@@ -89,7 +89,10 @@ def run(self):
 azure = ["azure-storage-blob==2.1.0"]
 oss = ["oss2==2.6.1"]
 ssh = ["paramiko>=2.5.0"]
-hdfs = ["pyarrow==0.15.1"]
+hdfs = [
+    "pyarrow==0.15.1; python_version < '3.8'",
```

Comment:
- [ ] Works with `0.15.1` for `Python < 3.8`.
- [ ] Works for `0.16.0` for `Python < 3.8`.

Reply:
- [ ] After ^ checks, should we force user to have `pyarrow==0.16.0` or keep as it is?